src/initialization.jl

```diff
@@ -282,7 +282,17 @@ function get_initial_values(prob, valp, f, alg::OverrideInit,
             throw(OverrideInitNoTolerance(:reltol))
         end
         nlsol = solve(initprob, nlsolve_alg; abstol = _abstol, reltol = _reltol, kwargs...)
-        success = SciMLBase.successful_retcode(nlsol)
+
+        success = if initprob isa NonlinearLeastSquaresProblem
+            # Do not accept StalledSuccess as a solution
+            # A good local minima is not a success 
+            resid = nlsol.resid
+            normresid = isdefined(integrator.opts, :internalnorm) ?
```

Comment:
@ChrisRackauckas `integrator` is not defined here. This causes test errors in ModelingToolkit, e.g. https://github.com/SciML/ModelingToolkit.jl/actions/runs/14966367552/job/42037129153?pr=3624

Reply:
Thanks for noticing. https://github.com/SciML/SciMLBase.jl/pull/1020 addresses this.

Reply:
ehh fuck. thanks for catching this